commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareReportServiceImpl.java

```diff
@@ -65,43 +65,33 @@ private ReportsMetadataResponseJson parseReportsFromResponse(String response) {
     }
 
     private List<ReportMetadataInfo> generateReportsFromReportsResponse(List<ReportMetadataJson> reportResponses) {
-        if (reportResponses == null) {
-            return Collections.emptyList();
-        }
-
-        List<ReportMetadataInfo> reportsInfoList = new ArrayList<>();
+        List<ReportMetadataInfo> reportsInfoList;
 
-        for (ReportMetadataJson reportResponse : reportResponses) {
-            reportsInfoList.add(populateReportMetadataInfo(reportResponse));
+        if (reportResponses == null) {
```

Comment:
this can be done better :
List<ReportMetadataInfo> reportsInfoList = Collections.emptyList();

if (reportResponses !=null) ...